pkg/restore/restore.go

```diff
@@ -1677,15 +1678,55 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs, itemExists
 	}
+
 	if patchBytes != nil {
-		if _, err = resourceClient.Patch(name, patchBytes); err != nil {
-			ctx.log.Errorf("error patch for managed fields %s: %v", kube.NamespaceAndName(obj), err)
-			if !apierrors.IsNotFound(err) {
-				errs.Add(namespace, err)
-				return warnings, errs, itemExists
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// First attempt to patch the object
+			if patchBytes != nil {
+				_, err = resourceClient.Patch(name, patchBytes)
 			}
-		} else {
-			ctx.log.Infof("the managed fields for %s is patched", kube.NamespaceAndName(obj))
+			if patchBytes == nil || err != nil {
+				ctx.log.Errorf("error patching managed fields %s: %v", kube.NamespaceAndName(obj), err)
+				if apierrors.IsConflict(err) {
```

Comment:
On the first execution of this func, Patch above may return a conflict, so this test is true -- and we regenerate and reapply patch. However, on subsequent executions, patchBytes is nil, so we don't have an error here, and therefore `IsConflict` is false. I think this should be `if err == nil || apierrors.IsConflict(err)`

Reply:
Won't the `RetryOnConflict` make this only on `apierrors.IsConflict(err)` already?

Reply:
It will retry if that was the prior error. I'm not 100% certain that the value of the var `err` will be set here, though. In any case, we know that any time this func is  called, if `err` is nil at this point, that means we didn't just patch above, so we want to enter this block, so I think it's clearer if we enter any `err` is nil or is a conflict.